### ISSUE TITLE
[BUG]Playwright recorder firing extra mouse events when interacting with BabylonJs canvas 

### ISSUE THREAD

**Issue by AnshPatel15** [microsoft/playwright#27926]
### System info
- Playwright Version: [1.38.1]
- Operating System: [Windows 10, macOS 14.0]
- Browser: [Chromium, Firefox]
- Other info:

### Source code

- [✅] I provided exact source code that allows reproducing the issue locally.

**Link to the GitHub repository with the repro**

https://github.com/AnshPatel15/babylonjs-playwright

- Since this is an issue with the recorder and not the script, I cant provide playwright code. But I have installed playwright into this babylonJs project that i have shared, with steps to run below. You can run and record using the VScode extension or codegen, and confirm the issue.

Video explaining the issue: https://www.loom.com/share/8727cda8af4b48a3b91ac3fb63338113

**Steps to recreate the issue*
- Install the above repo. clone and run "npm install". Then run "npm run dev". This should open the babylonjs canvas in a localhost server.
- since this is an issue with the playwright recorder, you will have to run the recorder (Im using the VScode extention but i could replicate this issue by using the command line as well)
- perform clicks on the canvas. theres a pointer down and a pointer up mouse event function in the app.jsx file
- when holding mouse down, it logs "mouse pressed" once. but when releasing the mouse, it logs "mouse released" and then again logs "mouse pressed" and another "mouse released". 
- This behaviour does not occur with the playwright recorder turned off. 

or

**Config file**

```js
module.exports = defineConfig({
  testDir: "./tests",
  /* Run tests in files in parallel */
  fullyParallel: true,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: process.env.CI ? 2 : 0,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 1 : undefined,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: "html",
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    /* Base URL to use in actions like `await page.goto('/')`. */
    // baseURL: 'http://127.0.0.1:3000',

    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: "on-first-retry",
  },
  timeout: 100000,

  /* Configure projects for major browsers */
  projects: [
    {
      name: "Draw Tests",
      use: { ...devices["Desktop Chrome"] },
      //testMatch : 'test data/draw',
      retries: 2,
      testDir: "src/tests/draw",
    },
    {
      name: "Edit Tests",
      use: { ...devices["Desktop Chrome"] },

      retries: 2,
      testDir: "src/tests/edit",
    },
    {
      name: "Indicator Tests",
      use: { ...devices["Desktop Chrome"] },

      retries: 2,
      testDir: "src/tests/indicators",
    },
  ],
```

**Expected**

The expected behaviour is that when recording on a babylonJS canvas, Playwright should not attach extra click events so that I can reliably record test cases. 

<img width="1000" alt="Screenshot 2023-11-02 at 6 01 43 PM" src="https://github.com/microsoft/playwright/assets/122334206/5478966d-5995-4dc3-8c4c-da6dfed75d2d">

console logs on a single click event

**Actual**

The interaction with the application should match with and without the recorder being turned on. These extra mouse events should not be added to the application while recording.

<img width="963" alt="Screenshot 2023-11-02 at 6 02 10 PM" src="https://github.com/microsoft/playwright/assets/122334206/2de675d0-6fbc-4ce2-8744-15b7c9b91bed">

2 mouse events added while recording on a single click event.

**Comment by AnshPatel15** [comment]
I wanted to ask, How can I confirm that this has been fixed? I tried reinstalling the latest Playwright npm package and I'm still getting the same bug

**Comment by dgozman** [comment]
@AnshPatel15 Take a look at our [canary releases guide](https://playwright.dev/docs/canary-releases). The fix should be available in tomorrow's canary release.

**Comment by AnshPatel15** [comment]
Thank you for replying so promptly. I shall let you know tomorrow after testing it out after release!

**Comment by haridas9890** [comment]
@dgozman, @pavelfeldman  Thank you for quick action and resolution on the issue.

**Comment by AnshPatel15** [comment]
@dgozman Hey, I want to point out that the double click issue is fixed, but any kind of drag functionality does not work now. so when recording, if i hold pointer down and move the box it does not move. This is adding problems in my other test cases. Can this be fixed as well?

**Comment by dgozman** [comment]
@AnshPatel15 Drag&drop is a know issue, not implemented yet, see #19892.

**Comment by AnshPatel15** [comment]
@dgozman I believe the above request is for the ability for codegen to generate code that is using drag. After this fix, im unable to use drag as a functionality on the webpage. which was previously possible.
